lib/coverband/utils/file_groups.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+#
+# Applies the configured groups to the given array of Coverband::SourceFile items
+#
+module Coverband
+  module Utils
+    class FileGroups
+      def initialize(files)
+        @grouped = {}
+        filter_to_groups(files)
```

Comment:
Thinking that if we set files as state in constructor, then might be able to extract more self documenting methods within this class and reduce the size of the filter_to_groups method. Something like this:

```ruby
def initialize(files)
  @grouped = {}
  @files = files
  filter_to_groups
end
```
Not sure..